packages/legend-studio/src/models/metamodels/pure/model/packageableElements/store/relational/connection/DatasourceSpecification.ts

```diff
@@ -266,10 +266,17 @@ export class BigQueryDatasourceSpecification
   }
 
   get hashCode(): string {
-    return hashArray([
-      CORE_HASH_STRUCTURE.BIGQUERY_DATASOURCE_SPECIFICATION,
-      this.projectId,
-      this.defaultDataset,
-    ]);
+    if (this.defaultDataset === undefined || this.defaultDataset === null) {
+      return hashArray([
```

Comment:
return hashArray([
        CORE_HASH_STRUCTURE.BIGQUERY_DATASOURCE_SPECIFICATION,
this.defaultDataset ?? '',
        this.projectId,
      ]);

Reply:
Right but in the typescript def, defaultDataset is required